src/functional/ffront/type_info.py

```diff
@@ -501,14 +558,20 @@ def function_signature_incompatibilities(
 
 @function_signature_incompatibilities.register
 def function_signature_incompatibilities_func(
-    func_type: ct.FunctionType, args: list[ct.SymbolType], kwargs: dict[str, ct.SymbolType]
+    func_type: ct.FunctionType,
+    args: list[ct.SymbolType],
+    kwargs: dict[str, ct.SymbolType],
 ) -> Iterator[str]:
 
     # check positional arguments
     if len(func_type.args) != len(args):
         yield f"Function takes {len(func_type.args)} argument(s), but {len(args)} were given."
     for i, (a_arg, b_arg) in enumerate(zip(func_type.args, args)):
-        if a_arg != b_arg and not is_concretizable(a_arg, to_type=b_arg):
+        if (
```

Comment:
The mechanism you implement in this PR is only valid for Field Operators, so it belongs into `function_signature_incompatibilities_fieldop`. Not sure how this will turn out, but I suggest to try an approach that resembles what one would do if one would implement this not on the code generation level, but in regular Python: If I get an argument that is a scalar and expect zero dimensional field, I convert the scalar into a zero dimensional field. Let me know if this is too cryptic and needs further explanation.

Reply:
the latter part of your suggestion is already done in `return_type_fieldop` as a general case for scalars